pytket_qir/qir.py

```diff
@@ -246,6 +299,10 @@ def block_to_circuit(self, block: QirBlock, circuit: Circuit) -> Circuit:
                 unitids = self.get_qubit_indices(instr.instr)
                 circuit.add_gate(op, unitids)
             elif instr.instr.is_qir_call:  # Setting the conditional bit for branching.
+                assert instr.instr.call_func_params
```

Comment:
More asserts -- are all these really impossible for a user to trigger? (Same goes for below; I won't comment further on this.)

Reply:
Actually these asserts are here to satisfy `mypy`. Most of these are actual Rust `Option` types that gets translated into Python `Optional` ones. Therefore, `mypy` complains that they could be `None` types if I haven't asserted for them.

Reply:
Done.